### ISSUE TITLE
Conflict when Node Typings are not installed

### ISSUE THREAD

**Issue by dstik** [apollographql/apollo-client#1247]
Hi Team,

Thank you again for the quick response time moving @types/node to devDependencies (per https://github.com/apollographql/apollo-client/issues/1078). 

I am working on a react-native project and have run into another issue where node typings are required.  When trying to call a GraphQL mutation, I get the following error from tsc.

```
1 /// <reference types="node" />
  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/apollo-client/core/QueryManager.d.ts(1,1): error TS2688: Cannot find type definition file for 'node'.


16         [queryId: string]: NodeJS.Timer | any;
                              ~~~~~~

node_modules/apollo-client/core/QueryManager.d.ts(16,28): error TS2503: Cannot find namespace 'NodeJS'.

```

Is there a workaround to use this with node typings (as they conflict with react-native typings)? 

For reference, my relevant code looks roughly like this:

```
import { graphql } from 'react-apollo';
import gql from 'graphql-tag';

const ComponentWithData = graphql(
  gql`
    mutation updateUser($user:UserAttributesInput!) {
      updateUser(user:$user) {
        id
        name
      }
    }
  `,
  {
    props: ({ mutate }) => ({
      submit: (userId: string, nameText: string) => mutate({
        variables: {
          employee: {
            id: userId,
            name: nameText,
          },
        },
      }),
    }),
  },
)(Component);
```

Thank you!
cc @wielrls

**Comment by calebmer** [comment]
The best thing to do here is to remove Node.js typings entirely. Which should just entail removing `NodeJS.Timer` in that one location as it has a union with `any` anyway. `T | any` will always just be `any` in TypeScript.

If you want to submit a PR it should be a simple change 👍

We’d keep `@types/node` as a dev dependency because it looks like we need them in tests, but otherwise, this reference can go as it appears to be the only thing in the non-test code that depends on `@types/node`.

**Comment by dstik** [comment]
Sounds good @calebmer - I will submit a PR. Thank you for the guidance and quick response!

**Comment by alejandronanez** [comment]
This was closed on #1248 right?

**Comment by calebmer** [comment]
Yep! Should be. Thanks for pointing it out @alejandronanez 😊
